controller/src/main/java/org/jboss/as/controller/ValidateModelStepHandler.java

```diff
@@ -43,7 +43,7 @@ private ValidateModelStepHandler() {
     @Override
     public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
         final Resource resource = loadResource(context);
-        if (resource == null) {
+        if (resource == null || resource.isRuntime()) {
```

Comment:
Perhaps it would be better to have the loadResource() method return null if a resource is runtime somewhere in the tree? Although it is a user error to define a non-runtime child of a runtime resource, I think it could still happen?